modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -529,10 +539,21 @@ INSFVMomentumAdvection::interpolate(Moose::FV::InterpMethod m, ADRealVectorValue
     mooseAssert(neighbor_a(i).value() != 0, "We should not be dividing by zero");
     neighbor_D(i) = neighbor_volume / neighbor_a(i);
   }
+
+  // We require this to ensure that the correct interpolation weights are used.
+  // This will change once the traditional weights are replaced by the weights
+  // that are used by the skewness-correction.
+  Moose::FV::InterpMethod face_velocity_interp_method =
+      (_advected_interp_method == Moose::FV::InterpMethod::SkewCorrectedAverage)
+          ? _advected_interp_method
+          : Moose::FV::InterpMethod::Average;
```

Comment:
Hmm it seems weird to have this tied to the advected interpolation method, but since the velocity interpolation method should always be Rhie-Chow, I think this should be fine for now. If we decide that we will in fact always do Rhie-Chow (which we probably should decide that), then perhaps we can change `velocity_interp_method` to have the logical meaning of "how do you want to interpolate the velocity before performing the RC correction to it"

Reply:
I agree, I think the Rhie-Chow interpolation falls into a different category. It is one layer above the average/vertex-based/skew-corrected. I'd like to work next year on a system which organizes these interpolation/limiter branches, I think the proposed modification can be a part of that.